packages/lib/src/components/styles/Link.tsx

```diff
@@ -0,0 +1,30 @@
+import { css } from "@emotion/react";
+import { Link, LinkProps } from "@leafygreen-ui/typography";
+import {
+  Link as RouterLink,
+  LinkProps as RouterLinkProps,
+} from "react-router-dom";
+
+// Override LeafyGreen's font-weight declaration for Link
+const overrideStyles = css`
+  font-weight: inherit;
+  font-size: inherit;
+  line-height: inherit;
+`;
+
+const StyledLink = (props: LinkProps<"a">) => (
+  // @ts-ignore-error: An internal LeafyGreen type causes this error.
+  <Link css={overrideStyles} hideExternalIcon {...props} />
+);
+
+const StyledRouterLink = (props: LinkProps<"span"> & RouterLinkProps) => (
+  // @ts-ignore-error: An internal LeafyGreen type causes this error.
```

Comment:
This may be preferable in case the error gets fixed
```suggestion
  // @ts-expect-error: An internal LeafyGreen type causes this error.
  <Link css={overrideStyles} hideExternalIcon {...props} />
);

const StyledRouterLink = (props: LinkProps<"span"> & RouterLinkProps) => (
  // @ts-expect-error: An internal LeafyGreen type causes this error.
```

Reply:
Okay I tried to change this, but it doesn't work unfortunately due to the TypeScript versions mismatching between `lib` and `apps`. lib uses `v5.1.3` and `apps` uses `v5.0.3` and they have a different opinion about what counts as an error. Tried to bump TypeScript but it caused a ton of errors 😞 I opened [DEVPROD-14004](https://jira.mongodb.org/browse/DEVPROD-14004)!